controllers/config_helper.go

```diff
@@ -0,0 +1,219 @@
+						"hypershift", "hive", "rhacs-operator", "open-cluster-*", "openshift-*",
+					},
+					Processes: []string{
+						"webhook", "mutation-webhook",
+					},
+				},
+			},
+		},
+	}
+
+	return config
+}
+
+func createDefaultConfig(ctx context.Context, c client.Client, namespace string,
+	gatekeeper *operatorv1alpha1.Gatekeeper, scheme *runtime.Scheme,
+) error {
+	var config *v1alpha1.Config
+
+	// When it is DisableDefaultMatches = true
+	if gatekeeper.Spec.Config != nil && gatekeeper.Spec.Config.DisableDefaultMatches {
+		config = &v1alpha1.Config{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "config",
+				Namespace: namespace,
+			},
+		}
+	} else {
+		config = getDefaultConfig(namespace)
+	}
+
+	matches := []v1alpha1.MatchEntry{}
+	if gatekeeper.Spec.Config != nil && gatekeeper.Spec.Config.Matches != nil {
+		// Will append gatekeeper.Spec.Config.Matches
+		matches = gatekeeper.Spec.Config.Matches
+	}
+
+	// Append matched from gatekeeper cr spec.config.matches
+	config.Spec.Match = append(config.Spec.Match, matches...)
+
+	// Set OwnerReference
+	if err := controllerutil.SetOwnerReference(gatekeeper, config, scheme); err != nil {
+		return err
+	}
+
+	err := c.Create(ctx, config)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Reset Config.Spec.Match and append the default exempt namespaces and
+// provided matches from the Gatekeeper CR
+func setExemptNamespaces(ctx context.Context,
+	c client.Client, existingConfig *v1alpha1.Config,
+	gatekeeper *operatorv1alpha1.Gatekeeper, scheme *runtime.Scheme, log logr.Logger,
+) error {
+	// Find OwnerReference
+	ownerRefFound := false
+
+	for _, ownerRef := range existingConfig.GetOwnerReferences() {
+		if ownerRef.UID == gatekeeper.UID {
+			ownerRefFound = true
+
+			break
+		}
+	}
+
+	// The ownerRefFound which is false means the Config resource was not created by gatekeeper-operator
+	if !ownerRefFound && len(existingConfig.Spec.Match) != 0 {
+		log.V(1).Info("The gatekeeper matches already exist. Skip adding DefaultExemptNamespaces")
+
+		return nil
+	}
+
+	// Reset existing Config Match
+	existingConfig.Spec.Match = nil
+
+	var configDefault *v1alpha1.Config
+
+	// When it is DisableDefaultMatches = false or nil then append default exempt namespaces
+	if gatekeeper.Spec.Config == nil || !gatekeeper.Spec.Config.DisableDefaultMatches {
+		configDefault = getDefaultConfig("")
+		existingConfig.Spec.Match = append(existingConfig.Spec.Match, configDefault.Spec.Match...)
+	}
+
+	// Avoid gatekeeper.Spec.Config nil error
+	if gatekeeper.Spec.Config != nil {
+		// Append matched from gatekeeper cr spec.config.matches
+		existingConfig.Spec.Match = append(existingConfig.Spec.Match, gatekeeper.Spec.Config.Matches...)
+	}
+
+	// Set OwnerReference
+	if !ownerRefFound {
+		if err := controllerutil.SetOwnerReference(gatekeeper, existingConfig, scheme); err != nil {
+			return err
+		}
+	}
+
+	err := c.Update(ctx, existingConfig, &client.UpdateOptions{})
```

Comment:
Could you guard against issuing the update if the config was already correct?

Reply:
Added: 
```
	if reflect.DeepEqual(existingConfig.Spec.Match, newMatch) {
		log.V(1).Info("No need to Update")

		return nil
	}
```